ricecooker/utils/pdf.py

```diff
@@ -71,8 +73,100 @@ def check_path(self):
         if not self.path:
             raise ValueError("self.path not found; call `open` first")
 
-
     def get_toc(self, subchapters=False):
+        temp_file = NamedTemporaryFile()
```

Comment:
Need to use `mkstemp` here - on windows trying to access a `NamedTemporaryFile` by referencing its name will give you a permissions error.

Also then need to explicitly close the file handle returned by mkstemp to ensure proper cleanup.

See here for an example: https://github.com/learningequality/ricecooker/blob/8cbdf5968457db8ea5d6d71f9d71f690bf8f0ad4/ricecooker/classes/files.py#L776

Reply:
> One update to start with - my suggestion in the issue was just to bite the bullet and use pdftk for all of our PDF operations, so we could drop reliance on the unmaintained Python library.

There is one issue here where we'll break the PDFParser API which I mentioned in my comment on the linked issue: https://github.com/learningequality/ricecooker/issues/312 - the `PDFParser.pdf` field is (in code) a public property. [It isn't mentioned in the docs though](https://ricecooker.readthedocs.io/en/latest/pdfutils.html). I'm okay with moving forward this way but feel it just needs to be noted that some peoples' chefs might break if they've managed to rely on that property at all - which they might because it is a PyPDF2 instance.

Will continue on with fully wrapping pdftk - but as we are effectively not doing anything to improve the software, I also suggested we consider just catching the error and informing the user of the limitations of PDFParser - which is that we cannot help them if their PDF doesn't have the proper metadata. This would be equally robust a solution if it can be worked out and ostensibly simpler. 

We have to do something like this anyway if we use pdftk.

Happy to go either way just wanted to hear your thoughts on the above.

Reply:
Also - its worth noting that the change I made is basically removing the meat of PyPDF2.

What remains is some clean-up and documentation updates to finalize the removal of PyPDF2. Also - I need to catch and raise a meaningful error message when we get a PDF without an outline.

Reply:
Yep - I think catching errors and giving meaningful error messages is an improvement :)